src/sentry/rules/processing/delayed_processing.py

```diff
@@ -13,24 +21,198 @@
+        if not isinstance(condition_inst, BaseEventFrequencyCondition):
+            logger.warning("Unregistered condition %r", condition_cls.id)
+            return None
+
+        _, duration = condition_inst.intervals[unique_condition.interval]
+        comparison_interval = condition_inst.intervals[unique_condition.interval][1]
+        comparison_type = (
+            condition_data.get("comparisonType", ComparisonType.COUNT)
+            if condition_data
+            else ComparisonType.COUNT
+        )
+        result = safe_execute(
+            condition_inst.get_rate_bulk,
+            duration,
+            comparison_interval,
+            group_ids,
+            unique_condition.environment_id,
+            comparison_type,
+        )
+        condition_group_results[unique_condition] = result
+    return condition_group_results
+
+
+def get_rules_to_fire(
+    condition_group_results: dict[UniqueCondition, dict[int, int]],
+    rule_to_slow_conditions: DefaultDict[Rule, list[MutableMapping[str, str] | None]],
+    rules_to_groups: DefaultDict[int, set[int]],
+) -> DefaultDict[Rule, set[int]]:
+    rules_to_fire = defaultdict(set)
+    for alert_rule, slow_conditions in rule_to_slow_conditions.items():
+        for slow_condition in slow_conditions:
+            if slow_condition:
+                condition_id = slow_condition.get("id")
+                condition_interval = slow_condition.get("interval")
+                target_value = int(str(slow_condition.get("value")))
+                for condition_data, results in condition_group_results.items():
+                    if (
+                        alert_rule.environment_id == condition_data.environment_id
+                        and condition_id == condition_data.cls_id
+                        and condition_interval == condition_data.interval
+                    ):
+                        for group_id in rules_to_groups[alert_rule.id]:
+                            if results[group_id] > target_value:
+                                rules_to_fire[alert_rule].add(group_id)
+    return rules_to_fire
+
+
 @redis_buffer_registry.add_handler(BufferHookEvent.FLUSH)
 def process_delayed_alert_conditions(buffer: RedisBuffer) -> None:
     with metrics.timer("delayed_processing.process_all_conditions.duration"):
         project_ids = buffer.get_set(PROJECT_ID_BUFFER_LIST_KEY)
 
         for project in RangeQuerySetWrapper(Project.objects.filter(id__in=project_ids)):
-            rulegroup_event_mapping = buffer.get_hash(
-                model=Project, field={"project_id": project.id}
-            )
-
             with metrics.timer("delayed_processing.process_project.duration"):
-                safe_execute(
-                    apply_delayed,
-                    project,
-                    rulegroup_event_mapping,
-                    _with_transaction=False,
-                )
+                apply_delayed.delay(project=project, buffer=buffer)
```

Comment:
We shouldn't pass the `project` here - this will use pickle and is generally recommended against. Just pass `project_id` and grab from postgres on the other side.

We also shouldn't pass `buffer` in here for the same reason. Either pass a string and have a way to look it up in the task, or just do `from sentry import buffer` in that task and use that directly

Reply:
Note to self: going to address this after it's merged